owncloudApp/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.kt

```diff
@@ -0,0 +1,2258 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.datamodel
+
+import android.accounts.Account
+import android.content.ContentProviderClient
+import android.content.ContentProviderOperation
+import android.content.ContentProviderResult
+import android.content.ContentResolver
+import android.content.ContentUris
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.OperationApplicationException
+import android.database.Cursor
+import android.net.Uri
+import android.os.Build
+import android.os.FileUriExposedException
+import android.os.RemoteException
+import android.provider.MediaStore
+
+import androidx.core.content.FileProvider
+import androidx.core.util.Pair
+import com.owncloud.android.MainApp
+import com.owncloud.android.R
+import com.owncloud.android.authentication.AccountUtils
+import com.owncloud.android.data.sharing.shares.db.OCShareEntity
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta
+import com.owncloud.android.domain.capabilities.model.CapabilityBooleanType
+import com.owncloud.android.domain.capabilities.model.OCCapability
+import com.owncloud.android.domain.sharing.shares.model.ShareType
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.shares.RemoteShare
+import com.owncloud.android.lib.resources.status.RemoteCapability
+import com.owncloud.android.utils.FileStorageUtils
+
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.util.ArrayList
+import java.util.Collections
+import java.util.HashSet
+import java.util.Vector
+
+class FileDataStorageManager {
+
+    var contentResolver: ContentResolver? = null
+        private set
+    var contentProviderClient: ContentProviderClient? = null
+        private set
+    var account: Account? = null
+    private var mContext: Context? = null
+
+    /**
+     * Get a collection with all the files set by the user as available offline, from all the accounts
+     * in the device, putting away the folders
+     *
+     *
+     * This is the only method working with a NULL account in [.mAccount]. Not something to do often.
+     *
+     * @return List with all the files set by the user as available offline.
+     */
+    // query for any favorite file in any OC account
+    val availableOfflineFilesFromEveryAccount: List<Pair<OCFile, String>>
+        get() {
+            val result = ArrayList<Pair<OCFile, String>>()
+
+            var cursorOnKeptInSync: Cursor? = null
+            try {
+                cursorOnKeptInSync = contentResolver!!.query(
```

Comment:
contentResolver can be null. If I use the other constructor and call this method I will get a NullPointerException.

Why don't you handle this cases better?